diff --git a/launch/nav2.launch.py b/launch/nav2.launch.py
index e91e6b0..ac2f256 100644
--- a/launch/nav2.launch.py
+++ b/launch/nav2.launch.py
@@ -23,8 +23,8 @@ def generate_launch_description():
 
     robot = LaunchArgument('robot')
     task_generator_node = LaunchArgument('task_generator_node')
-    namespace = LaunchArgument('namespace')
-    frame = LaunchArgument('frame')
+    namespace = LaunchArgument('namespace', default_value='')
+    frame = LaunchArgument('frame', default_value='')
     use_sim_time = LaunchArgument('use_sim_time')
     global_planner = LaunchArgument('global_planner')
     local_planner = LaunchArgument('local_planner')
@@ -174,7 +174,7 @@ def generate_launch_description():
         ('map_server', '/map_server'),
         ('/tf', '/tf'),
         ('/tf_static', '/tf_static'),
-        ('map', PathJoinSubstitution([task_generator_node.substitution, 'map'])),
+        ('map', PathJoinSubstitution(['/', task_generator_node.substitution, 'map'])),
     ]
 
     bringup_cmd_group = GroupAction([
@@ -184,6 +184,9 @@ def generate_launch_description():
             executable='relay',
             name='goal_pose_relay',
             arguments=['/goal_pose', 'goal_pose'],
+            condition=IfCondition(PythonExpression([
+                "'", namespace.substitution, "' != ''"
+            ])),
         ),
         # Node(
         #     package='tf2_ros',
